src/AspNet.Security.OAuth.Alipay/AlipayAuthenticationHandler.cs

```diff
@@ -0,0 +1,217 @@
+            }
+
+            using var stream = await response.Content.ReadAsStreamAsync(Context.RequestAborted);
+            using var document = JsonDocument.Parse(stream);
+
+            var mainElement = document.RootElement.GetProperty("alipay_system_oauth_token_response");
+            if (!ValidateReturnCode(mainElement, out var code))
+            {
+                return OAuthTokenResponse.Failed(new Exception($"An error (Code:{code}) occurred while retrieving an access token."));
+            }
+
+            var payload = JsonDocument.Parse(mainElement.GetRawText());
+            return OAuthTokenResponse.Success(payload);
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync(
+            [NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties,
+            [NotNull] OAuthTokenResponse tokens)
+        {
+            // See https://opendocs.alipay.com/apis/api_2/alipay.user.info.share for details.
+            var sortedParams = new SortedDictionary<string, string>()
+            {
+                ["app_id"] = Options.ClientId,
+                ["auth_token"] = tokens.AccessToken,
+                ["charset"] = "utf-8",
+                ["format"] = "JSON",
+                ["method"] = "alipay.user.info.share",
+                ["sign_type"] = "RSA2",
+                ["timestamp"] = Clock.UtcNow.ToString("yyyy-MM-dd HH:mm:ss", CultureInfo.InvariantCulture),
+                ["version"] = "1.0",
+            };
+            sortedParams.Add("sign", GetRSA2Signature(sortedParams));
+
+            string address = QueryHelpers.AddQueryString(Options.UserInformationEndpoint, sortedParams!);
+
+            using var response = await Backchannel.GetAsync(address, Context.RequestAborted);
+
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync());
+
+                throw new HttpRequestException("An error occurred while retrieving user information.");
+            }
+
+            using var stream = await response.Content.ReadAsStreamAsync(Context.RequestAborted);
+            using var document = JsonDocument.Parse(stream);
+            var rootElement = document.RootElement;
+
+            if (!rootElement.TryGetProperty("alipay_user_info_share_response", out var mainElement))
+            {
+                string errorCode = rootElement.GetProperty("error_response").GetProperty("code").GetString() !;
```

Comment:
nit: please `var` and remove unnecessary spaces before `!` (here and elsewhere)

Reply:
(The `{space}!` is a StyleCop workaround)

Reply:
Crap... 😢 
Thanks for the info!

Reply:
Done.